### PR TITLE
Refs [TASK][26172] bind real data

### PR DESCRIPTION
**Pull request by aug16vcc**
## Related Tickets
- [#Ticket ID](https://edu-redmine.sun-asterisk.vn/issues/26172)

## WHAT
- Write Search fragment ui

## HOW
- Connect repo - presenter
## WHY
- Important

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Vu Chi Cong | Reviewer2 (HuyTQ)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [x] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [x] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/boringweather/util/PreferenceUtils.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        context.defaultSharedPreferences?.getInt(
+            context.getString(R.string.pref_key_selected_city_id),
+            default
+        ) ?: default
+
+    fun setSelectedCityId(context: Context, value: Int) =
+        context.defaultSharedPreferences?.edit {
+            putInt(context.getString(R.string.pref_key_selected_city_id), value)
+        }
+}
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/boringweather/util/PreferenceUtils.kt]: sao ông không viêt dưới dạng extentions của SharedPreferences nhỉ 
t thấy  context.defaultSharedPreferences lặp lại nhiều

File: app/src/main/java/com/sunasterisk/boringweather/util/PreferenceUtils.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/boringweather/util/PreferenceUtils.kt]: thừa dòng trống

File: app/src/main/java/com/sunasterisk/boringweather/util/UnitSystem.kt [hunk 3]
```diff
@@ -2,45 +2,48 @@ package com.sunasterisk.boringweather.util
...
+        METRIC -> R.string.format_speed_metric to speed
+        IMPERIAL -> R.string.format_speed_imperial to speed * Constants.METER_PER_SECOND_TO_MILES_PER_HOUR
+        INTERNATIONAL -> R.string.format_speed_international to speed
+    }.let { (stringRes, unitSpeed) -> resources.getString(stringRes, unitSpeed) }
 
     fun formatPressure(pressure: Int, resources: Resources) =
         resources.getString(
             R.string.format_pressure,
-            pressure.toFloat() / Constants.PRESSURE_EXCHANGE_RATIO
+            pressure.toFloat() / KILO_PA_TO_HECTO_PA
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/boringweather/util/UnitSystem.kt]: kilo pato hecto n là cái gì thế ?

File: app/src/main/java/com/sunasterisk/boringweather/util/PreferenceUtils.kt (hunk 1 above)
Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/boringweather/util/PreferenceUtils.kt]: Thank you

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentPresenter.kt [hunk 4]
```diff
@@ -16,28 +16,27 @@ class CurrentPresenter(
     override fun loadCityById(cityId: Int) {
         cityRepository.getCityById(cityId) {
             when (it) {
-                is Result.Success -> it.data.let(view::showCity)
+                is Result.Success -> it.data.let { city ->
+                    view.showCity(city)
+                    refreshCurrentWeather(city, false)
+                }
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentPresenter.kt]: liệu cần ngoặc kép k nhỉ

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt [hunk 5]
```diff
@@ -54,23 +69,35 @@ class CurrentFragment : BaseFragment(), CurrentContract.View,
...
         }
 
+        toolbarSearch.setOnMenuItemClickListener {
+            when (it.itemId) {
+                R.id.searchCity -> {
+                    findNavigator()?.navigateToSearchFragment()
+                    true
+                }
+                else -> false
+            }
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt]: check có mỗi một cái mà vẫn dùng when à ?

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt [hunk 6]
```diff
@@ -54,23 +69,35 @@ class CurrentFragment : BaseFragment(), CurrentContract.View,
     private fun initView() {
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt]:
```suggestion
    private fun initViews() {
```

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt [hunk 7]
```diff
@@ -42,6 +49,14 @@ class CurrentFragment : BaseFragment(), CurrentContract.View,
             Injector.getOneCallRepository(requireContext()),
             Injector.getCityRepository(requireContext())
         )
+
+        // is lifecycle aware listener
+        setFragmentResultListener(SearchConstants.KEY_REQUEST_SEARCH_CITY) { requestKey: String, bundle: Bundle ->
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt]: t thấy ô hay lạm dụng comment . hơn nữa comment trên là IS mà dưới hàm là SET, nhìn qua thấy k hợp lý lắm

> Don’t Use a Comment When You Can Use a Function or a Variable

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt [hunk 8]
```diff
@@ -6,14 +6,19 @@ import android.view.View
 import android.widget.Toast
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt:6]: Hình như ô dùng showToast ở Util r thì  ở đây vẫn import Toast hơi thừa phải k. ô thử check lại xem

File: app/src/main/java/com/sunasterisk/boringweather/service/PrePopulateDatabaseService.kt [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+            val url = URL(urlString)
+            val connection = url.openConnection()
+            connection.connect()
+            val inStream = connection.getInputStream()
+            val sql = inStream.bufferedReader().readText()
+            AppDatabase.getInstance(this@PrePopulateDatabaseService)
+                .writableDatabase.execSQL(sql)
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/boringweather/service/PrePopulateDatabaseService.kt]: chỉ in mỗi ra exception mà k xử lý gì nhỉ ?

File: app/src/main/java/com/sunasterisk/boringweather/util/UnitSystem.kt (hunk 3 above)
Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/boringweather/util/UnitSystem.kt]: Là kPa sang hPa

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentPresenter.kt (hunk 4 above)
Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentPresenter.kt]: T chưa hiểu lắm ông có thể suggest được không

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt (hunk 7 above)
Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt]: Sorry t viết xong đoạn sau quên k xóa

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt (hunk 8 above)
Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentFragment.kt:6]: Là import để truyền vào `Toast.LENGTH_LONG` ô ạ

File: app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentPresenter.kt (hunk 4 above)
Comment (thonc-2135) [reply app/src/main/java/com/sunasterisk/boringweather/ui/current/CurrentPresenter.kt]: kia hàm dài hơn 1 dòng thì có cần ngoặc kép bao ngoài k, ô đang k dùng mà

File: app/src/main/java/com/sunasterisk/boringweather/ui/splash/SplashActivity.kt [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+        if (defaultSharedPreferences.isFirstLaunch) {
+            PrePopulateDatabaseService.enqueueWork(
+                this, Intent().putExtra(
+                    Constants.EXTRA_URL_STRING_PREPOPULATE_DATABASE_SERVICE, getString(
+                        R.string.url_prepopulate_city_table_sql
+                    )
+                )
+            )
+        } else navigateToMainActivity()
```
Comment (datdt-1935) [inline app/src/main/java/com/sunasterisk/boringweather/ui/splash/SplashActivity.kt]:
```suggestion
        } else {
            navigateToMainActivity()
        }
```

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -18,11 +18,11 @@
     <string name="title_search_city">Search city</string>
 
     <string name="title_holder_percent">-- %</string>
-    <string name="title_holder_distance">-- Km</string>
-    <string name="title_holder_speed">-- m/s</string>
+    <string name="title_holder_distance">--</string>
+    <string name="title_holder_speed">--</string>
```
Comment (datdt-1935) [inline app/src/main/res/values/strings.xml]: mấy cái này khác gì nhau vậy em :|

**Pull request merged**